hubspot3/test/test_tickets.py

```diff
@@ -49,3 +57,207 @@ def test_get_all_tickets():
+                        'source': 'CALCULATED',
+                        'sourceVid': []
+                    }],
+                    'value': '1571427728000',
+                    'timestamp': 1571411169418,
+                    'source': 'CALCULATED',
+                    'sourceId': 'TicketsRollupProperties'
+                },
+                'hs_last_email_activity': {
+                    'versions': [{
+                        'name': 'hs_last_email_activity',
+                        'value': 'SENT_TO_CONTACT',
+                        'timestamp': 1571411169118,
+                        'sourceId': 'TicketsRollupProperties',
+                        'source': 'CALCULATED',
+                        'sourceVid': []
+                    }, {
+                        'name': 'hs_last_email_activity',
+                        'value': 'REPLY_FROM_CONTACT',
+                        'timestamp': 1571409899877,
+                        'sourceId': 'TicketsRollupProperties',
+                        'source': 'CALCULATED',
+                        'sourceVid': []
+                    }],
+                    'value': 'SENT_TO_CONTACT',
+                    'timestamp': 1571411169118,
+                    'source': 'CALCULATED',
+                    'sourceId': 'TicketsRollupProperties'
+                }
+            }
+        }
+    }
+    # TODO should add more complex data to test, this is not testing the full complexity of the method
+    ids = [47005994]
+    properties = ['hs_lastcontacted', 'hs_last_email_activity']
+    params_batch = {"propertiesWithHistory": p for p in properties}
+    params_recent = {
+        'objectId': 47005994,
+        'timestamp': 1571411169000,
+        'changeType': changeType
+    }
+
+    mock_connection.set_responses([(200, json.dumps(response_body_recent)),
+                                   (200, json.dumps(response_body_batch)),
+                                   (200, '[]')])
+    if changeType == 'CHANGED':
+        changes = tickets_client.get_recently_modified(time_offset=1571847002)
+    else:
+        changes = tickets_client.get_recently_created(time_offset=1571847002)
+    mock_connection.assert_num_requests(3)
+
+    mock_connection.assert_has_request(
+        "GET", "/crm-objects/v1/change-log/ticket", **params_recent
+    )
+    mock_connection.assert_has_request(
+        "POST", "/crm-objects/v1/objects/tickets/batch-read", **params_batch, data={'ids': ids},
+    )
+    assert len(changes) == 2
+    assert [change['objectId'] for change in changes] == [change['objectId'] for
+                                                          change in response_body_recent]
+    assert len(changes[0]['changes']['changedValues']) == 1
+    assert len(changes[1]['changes']['changedValues']) == 2
+
+
+def test_get_recently_created(
+    tickets_client,
+    mock_connection,
+):
+    base_get_recently(tickets_client, mock_connection, 'CREATED')
```

Comment:
Instead of passing a constant 'CREATED' or 'CHANGED' and write a `if` for it, you can pass a function that given a `ticket_client` returns the correct method. This way you avoid that if entirely.

Reply:
the problem is that changeType is used to check if the request was made correctly, so I would have to pass them both